app/controllers/courses_controller.rb

```diff
@@ -4,8 +4,9 @@ class CoursesController < ApplicationController
 
   def index
     @q = Course.ransack(params[:q])
-    @courses = CoursesQuery.new(relation: @q.result(distinct: true).includes(:tags),
-                                params: filter_params).call.decorate
+    query = CoursesQuery.new(relation: @q.result(distinct: true).includes(:tags), params: filter_params,
```

Comment:
Move ransack search to Query object:

query = CoursesQuery.new(relation: permitted_cources, params: filter_params, student: current_student).call